rust/blockstore/src/memory/reader_writer.rs

```diff
@@ -122,11 +130,18 @@ impl<
         }
         let values = values
             .iter()
-            .map(|(key, value)| (key.prefix.as_str(), K::from(&key.key), value.clone()))
+            .map(|(key, value)| {
+                (
+                    key.prefix.as_str(),
+                    K::try_from(&key.key).unwrap(),
```

Comment:
If this unwrap fails what happens?

Reply:
Same thing that happened with the panic it replaces :-(